arm-mediaservices/2015-10-01/swagger/media.json

```diff
@@ -301,29 +338,52 @@
               "$ref": "#/definitions/MediaService"
             }
           }
-        ]
-      }
-    },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Media/mediaservices/{mediaServiceName}/regenerateKey": {
-      "post": {
-        "summary": "Regenerate the key for a Media Service.",
-        "operationId": "MediaService_RegenerateKey",
+        ],
         "responses": {
           "200": {
-            "description": "OK",
+            "description": "Success. The Media Services account was updated.",
             "schema": {
-              "$ref": "#/definitions/RegenerateKeyOutput"
+              "$ref": "#/definitions/MediaService"

```

Comment:
- A PATCH operation cannot have a model that has required properties and the properties cannot have default values. Because PATCH is like a diff. The customer is supposed to provide only the properties that he/she want to update. 
- What kind of update does the PATCH operation support?
   - Does it only support updating tags? If yes then there is no value in exposing PATCH as that can be done by PUT as well. (I assume a second call to PUT will not fail and the service will replace the object with the new object provided in the body parameter in that request, correct?)
  - Can the user update the list of storageAccounts in PATCH? If yes then how is that achieved? Will the service replace the old array with the new one provided in PATCH? (Just confirming out of curiosity)